File: Java/Solutions/4.2. Complex Conditions - Exam Problems/OnTimeForTheExam.java
import java.lang.*;
import java.util.Scanner;

public class OnTimeForTheExam {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        // Set some constants.
        String late = "Late";
        String onTime = "On time";
        String early = "Early";

        // Read the input.
        int examHours = Integer.parseInt(scanner.nextLine());
        int examMinutes = Integer.parseInt(scanner.nextLine());
        int arrivalHours = Integer.parseInt(scanner.nextLine());
        int arrivalMinutes = Integer.parseInt(scanner.nextLine());

        // Calculate all differences in minutes.
        int examTime = (examHours * 60) + examMinutes;
        int arrivalTime = (arrivalHours * 60) + arrivalMinutes;
        int totalMinutesDifference = arrivalTime - examTime;

        // Check when the student had arrived for the exam and display the result
        String studentArrival = late;
        if (totalMinutesDifference < -30) {
            studentArrival = early;
        } else if (totalMinutesDifference <= 0) {
            studentArrival = onTime;
        }

        // Check and output final result.
        String result = "";
        if (totalMinutesDifference != 0) {
            int hoursDifference = Math.abs(totalMinutesDifference / 60);
            int minutesDifference = Math.abs(totalMinutesDifference % 60);

            if (hoursDifference > 0) {
                result = String.format("%d:%02d hours", hoursDifference, minutesDifference);
            } else {
                result = minutesDifference + " minutes";
            }

            if (totalMinutesDifference < 0) {
                result += " before the start";
            } else {
                result += " after the start";
            }
        }

        System.out.println(studentArrival);
        if (!result.isEmpty()) {
            System.out.println(result);
        }
    }
}

File: CSharp/Solutions/4.2. Complex Conditions - Exam Problems/OnTimeForTheExam.cs
namespace OnTimeForTheExam
{
    using System;

    public static class OnTimeForTheExam
    {
        public static void Main()
        {
            // Set some constants.
            string late = "Late";
            string onTime = "On time";
            string early = "Early";

            // Read the input.
            int examHours = int.Parse(Console.ReadLine());
            int examMinutes = int.Parse(Console.ReadLine());
            int arrivalHours = int.Parse(Console.ReadLine());
            int arrivalMinutes = int.Parse(Console.ReadLine());

            // Calculate all diffences in minutes.
            int examTime = (examHours * 60) + examMinutes;
            int arrivalTime = (arrivalHours * 60) + arrivalMinutes;
            int totalMinutesDifference = arrivalTime - examTime;

            // Check when the student had arrived for the exam and display the result
            string studentArrival = late;
            if (totalMinutesDifference < -30)
            {
                studentArrival = early;
            }
            else if (totalMinutesDifference <= 0)
            {
                studentArrival = onTime;
            }

            // Check and output final result.
            string result = string.Empty;
            if (totalMinutesDifference != 0)
            {
                int hoursDifference = Math.Abs(totalMinutesDifference / 60);
                int minutesDifference = Math.Abs(totalMinutesDifference % 60);

                if (hoursDifference > 0)
                {
                    result = string.Format("{0}:{1:00} hours", hoursDifference, minutesDifference);
                }
                else
                {
                    result = minutesDifference + " minutes";
                }

                if (totalMinutesDifference < 0)
                {
                    result += " before the start";
                }
                else
                {
                    result += " after the start";
                }
            }

            Console.WriteLine(studentArrival);
            if (!string.IsNullOrEmpty(result))
            {
                Console.WriteLine(result);
            }
        }
    }
}
